src/main/scala/cromwell/binding/types/WdlFileGlobType.scala

```diff
@@ -0,0 +1,33 @@
+package cromwell.binding.types
+
+import cromwell.binding.values.{WdlFile, WdlFileGlob, WdlString}
+import spray.json.JsString
+
+import scala.util.{Success, Try}
+
+// TODO: Do we even need this?
+
+/**
+ * This is never a type which gets evaluated to - we get them from the FileEvaluator for tracking "glob" functions
+ * in output declarations.
+ */
+case object WdlFileGlobType extends WdlPrimitiveType {
+  val toWdlString: String = "FileGlob"
+
+  override protected def coercion = {
+    case s: String => WdlFileGlob(s)
+    case s: JsString => WdlFileGlob(s.value)
+    case s: WdlString => WdlFileGlob(s.valueString)
+    case f: WdlFile => WdlFileGlob(f.valueString)
+  }
+
+  override def add(rhs: WdlType): Try[WdlType] = rhs match {
+    case _ => invalid(s"$this + $rhs")
+  }
+
+  override def equals(rhs: WdlType): Try[WdlType] = rhs match {
```

Comment:
Woah, good catch @mcovarr.

One possibility is to utilize scalaz's Equals typeclass but that might be an equally as horrible an idea